packages/@aws-cdk/aws-lambda-event-sources/lib/sqs.ts

```diff
@@ -29,14 +39,20 @@ export class SqsEventSource implements lambda.IEventSource {
   private _eventSourceMappingId?: string = undefined;
 
   constructor(readonly queue: sqs.IQueue, private readonly props: SqsEventSourceProps = { }) {
-    if (this.props.batchSize !== undefined && (this.props.batchSize < 1 || this.props.batchSize > 10)) {
-      throw new Error(`Maximum batch size must be between 1 and 10 inclusive (given ${this.props.batchSize})`);
+    if (this.props.batchSize !== undefined) {
```

Comment:
Looks like we've missed to check for tokens here. Could you add that, and a corresponding unit test?

Reply:
```suggestion
    if (this.props.batchSize !== undefined && !Token.isUnresolved(this.props.batchSize)) {
```